napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -0,0 +1,255 @@
+        layout.addLayout(hlayout2)
+        layout.addWidget(self._table)
+
+        self.setLayout(layout)
+
+    def _set_table(self, layer_str=''):
+        if layer_str == '':
+            layer_str = self.ALL_ACTIVE_KEYBINDINGS
+
+        try:
+            self._table.cellChanged.disconnect(self._set_keybinding)
+        except TypeError:
+            pass
+
+        self._table.clearContents()
+
+        self._table.horizontalHeader().setStretchLastSection(True)
+        self._table.horizontalHeader().setSectionResizeMode(
+            QHeaderView.Stretch
+        )
+
+        actions = self.key_bindings_strs[layer_str]
+
+        if len(actions) > 0:
+
+            self._table.setRowCount(len(actions))
+            # self._table.setRowCount(len(action_manager._actions))
+            self._table.setColumnCount(3)
+            self._table.setHorizontalHeaderLabels(['Action', 'Keybinding'])
+            self._table.verticalHeader().setVisible(False)
+
+            self._table.setColumnHidden(2, True)
+
+            for row, (action_name, action) in enumerate(actions.items()):
+                shortcuts = action_manager._shortcuts.get(action_name, [])
+                item = QTableWidgetItem(action.description)
+
+                item.setFlags(item.flags() & ~Qt.ItemIsEditable)
+
+                self._table.setItem(row, 0, item)
+                item_shortcut = QTableWidgetItem(
+                    list(shortcuts)[0] if shortcuts else ""
+                )
+
+                self._table.setItem(row, 1, item_shortcut)
+
+                item_action = QTableWidgetItem(action_name)
+                # action_name is stored in the table to use later, but is not shown on dialog.
+                self._table.setItem(row, 2, item_action)
+
+            self._table.cellChanged.connect(self._set_keybinding)
+        else:
+            # no actions-- display that.
+            self._table.setRowCount(1)
+            self._table.setColumnCount(3)
+            self._table.setHorizontalHeaderLabels(['Action', 'Keybinding'])
+            self._table.verticalHeader().setVisible(False)
+
+            self._table.setColumnHidden(2, True)
+            item = QTableWidgetItem('No key bindings')
+            item.setFlags(item.flags() & ~Qt.ItemIsEditable)
+            self._table.setItem(0, 0, item)
+
+    def _set_keybinding(self, event):
+        # event is the index of the event.
+
+        current_layer_text = self.layer_combo_box.currentText()
+        layer_actions = self.key_bindings_strs[current_layer_text]
+
+        # get the current item from shortcuts column
+        item1 = self._table.currentItem()
+        new_shortcut = item1.text()
+
+        # get the action name
+        current_action = self._table.item(event, 2).text()
+
+        replace = True
+        for row, (action_name, action) in enumerate(layer_actions.items()):
+            shortcuts = action_manager._shortcuts.get(action_name, [])
+
+            if new_shortcut in shortcuts:
```

Comment:
IIRC< That will not be totally correct you will have shortcuts that conflicts across all layers. Typically Z is use to pan zoom in all layers, 
So we need to check if the shortcut exists in actions with the same Keymapprovider in the long term.

Reply:
For pan zoom, it is in all layers, but I am seeing that each of the actions for pan zoom has a different keymapprovider for each layer.  
Here is a list of the keymapproviders for pan_zoom for each layer.  There isn't one for the viewer.

```
<class 'napari.layers.labels.labels.Labels'> napari:activate_label_pan_zoom_mode
<class 'napari.layers.points.points.Points'> napari:activate_points_pan_zoom_mode
<class 'napari.layers.shapes.shapes.Shapes'> napari:activate_shape_pan_zoom_mode
```

I realize a person could overwrite the zoom for one layer, and it will still be `Z' for the others, but I guess they could do that if they wanted?